docs/advanced/language-extensions.md

```diff
@@ -366,11 +366,31 @@ end
 
 ### `LuaMap` and `LuaSet`
 
-Similar to `LuaTable`, the `LuaMap` and `LuaSet` types are provided to represent Lua tables used as a map or set. These are much more performant than the `Set`/`Map` classes, but do not come with all the same features (such as guaranteed insertion order).
+Similar to `LuaTable`, the `LuaMap` and `LuaSet` types are provided to represent Lua tables that are used as a map or a set. These are more performant than the `Set`/`Map` classes, but do not come with all of the same features (such as guaranteed insertion order).
 
-- `LuaMap` has the same methods as `LuaTable`, except that `table.get(key)` may also return `undefined`.
-- `LuaSet`, instead of `table.get/table.set`, has `table.add(value)`, which translates to `table[value] = true`.
-- There are also the readonly variants `ReadonlyLuaMap` and `ReadonlyLuaSet`.
+- `LuaMap` has the same methods as `LuaTable`. The exception is that the type of the return value for the `get` method is `V | undefined` instead of `V`, which makes it similar to how a `Map` works. (For this reason, we recommend using `LuaMap` over `LuaTable`, as it provides compiler-safety guarantees whenever you access a value.)
+- `LuaSet` does not have `get` and `set` methods. Instead, it has the `add` method, which transpiles to `table[value] = true`.
+- TSTL also provides the read-only variants of `ReadonlyLuaMap` and `ReadonlyLuaSet`, if needed.
+
+#### Iterating over `LuaMap` & `LuaSet`
+
+If you need to iterate over a `LuaMap` or `LuaSet`, you do so "implicitly", in the same way that you would for `LuaTable`:
```

Comment:
I don't like the phrasing that this is an `implicit` iteration. `Map`/`Set` in Javascript are also iterated the same way.